File: python3/Q682_baseball_game.py
class Solution:
    def calPoints(self, operations: List[str]) -> int:
        scores = []

        for op in operations:
            if op.lstrip('-').isdigit():
                scores.append(int(op))
            elif op == '+':
                scores.append(scores[-1] + scores[-2])
            elif op == 'D':
                scores.append(scores[-1] * 2)
            elif op == 'C':
                scores.pop()

        return sum(scores)

File: java/Q682_baseball_game.java
import java.util.Stack;

public class Q682_baseball_game {
    public int calPoints(String[] operations) {
        if (operations == null || operations.length == 0) return 0;
        Stack<Integer> stack = new Stack<>();
        int sum = 0;

        for (String op: operations) {
            if (op.equals("+")) {
                if (stack.size() >= 2) {
                    int e1 = stack.pop();
                    int e2 = stack.peek();
                    int twoSum = e1 + e2;
                    sum += twoSum;
                    stack.push(e1);
                    stack.push(twoSum);
                }
            } else if (op.equals("D")) {
                if (!stack.isEmpty()) {
                    int doubledScore = stack.peek() * 2;
                    sum += doubledScore;
                    stack.push(doubledScore);
                }
            } else if (op.equals("C")) {
                if (!stack.isEmpty()) {
                    int lastScore = stack.pop();
                    sum -= lastScore;
                }
            } else {
                int now = Integer.parseInt(op);
                sum += now;
                stack.push(now);
            }
        }
        
        return sum;
    }
}